pkgs/top-level/all-packages.nix

```diff
@@ -6729,12 +6729,12 @@ in
 
   opensm = callPackage ../tools/networking/opensm { };
 
-  openssh =
-    callPackage ../tools/networking/openssh {
-      hpnSupport = false;
-      etcDir = "/etc/ssh";
-      pam = if stdenv.isLinux then pam else null;
-    };
+  inherit (callPackages ../tools/networking/openssh {
+    hpnSupport = false;
+    etcDir = "/etc/ssh";
+    pam = if stdenv.isLinux then pam else null;
```

Comment:
This is not the cleanest way because packages shouldn't be overwritten with null. I would prefer a withPam or enablePam flag.

Reply:
I added an `withPam` option.